docs/DeveloperGuide.md

```diff
@@ -1,13 +1,71 @@
 # Developer Guide
 
+- [Acknowledgements](#acknowledgements)
+- [Design](#design)
+  - [Architecture](#architecture)
+- [Implementation](#implementation)
+  - [`add` feature](#add-feature)
+  - [`delete` feature](#delete-feature)
+  - [`set` spending limit feature](#set-spending-limit-feature)
+  - [`list` feature](#list-feature)
+  - [`help` feature](#help-feature)
+  - [[Proposed] `find` feature](#proposed-find-feature)
+- [Product scope](#product-scope)
+  - [Target user profile](#target-user-profile)
+  - [Value proposition](#value-proposition)
+- [User stories](#user-stories)
+- [Non-functional requirements](#non-functional-requirements)
+- [Glossary](#glossary)
+- [Instructions for manual testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
-
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
-
+## Design
+### Architecture
+![](Architecture Diagram.jpg)
```

Comment:
Architecture diagram seems a little atypical ( at least compared to what's presented in https://nus-cs2103-ay1920s2.github.io/website/se-book-adapted/chapters/architecture.html.). Also would be better to explain briefly on diagram how the components are connected


![image](https://user-images.githubusercontent.com/47512610/160960643-b1f632a9-02b6-4749-96b2-879e69b20da9.png)